test_runner/src/main/kotlin/ftl/ios/Xctestrun.kt

```diff
@@ -70,41 +79,55 @@ object Xctestrun {
         return PropertyListParser.parse(xctestrun) as NSDictionary
     }
 
-    fun findTestNames(xctestrun: String): List<String> {
+    fun findTestNames(xctestrun: String): XctestrunMethods {
         return findTestNames(File(xctestrun))
     }
 
-    // Finds tests in a xctestrun file
-    private fun findTestNames(xctestrun: File): List<String> {
-        val root = parse(xctestrun)
-        val result = mutableListOf<String>()
-        // EarlGreyExampleSwiftTests_iphoneos11.3-arm64.xctestrun => EarlGreyExampleSwiftTests
-        val testRoot = xctestrun.parent + "/"
-
-        // OnlyTestIdentifiers
-        // Test-Class-Name[/Test-Method-Name]
-        // Example/testExample
-        for (testTarget in root.allKeys()) {
-            val testDictionary = (root[testTarget] as NSDictionary)
-
-            val methods = testsForTarget(
-                testDictionary = testDictionary,
-                testRoot = testRoot,
-                testTarget = testTarget
+    fun findTestNames(testTarget: String, xctestrun: String): List<String> =
+        findTestNamesForTarget(
+            testTarget = testTarget,
+            xctestrun = File(xctestrun)
+        )
+
+    /* Finds tests in a xctestrun file */
+    private fun findTestNames(xctestrun: File): XctestrunMethods =
+        parse(xctestrun).allKeys().associate { testTarget ->
+            testTarget to findTestNamesForTarget(
+                testTarget = testTarget,
+                xctestrun = xctestrun
             )
-
-            result += methods
         }
 
-        return result.distinct()
+    /* Finds tests for testTarget in xctestrun file */
+    private fun findTestNamesForTarget(
+            testTarget: String,
+            xctestrun: File
+    ): List<String> =
+        testsForTarget(
+            testDictionary = parse(xctestrun)[testTarget]
+                as? NSDictionary
+                ?: throw FlankGeneralError("XCTestrun does not contain $testTarget test target."),
+            testRoot = xctestrun.parent + "/",
+            testTarget = testTarget
+        ).distinct()
+
+    fun rewrite(xctestrun: String, methods: List<String>): ByteArray {
+        val xctestrunFile = File(xctestrun)
+        val methodsToRun = findTestNames(xctestrunFile).mapValues { it.value.filter { methods.contains(it) } }
```

Comment:
I suggest suggestions like that ;)
```suggestion
        val methodsToRun = findTestNames(xctestrunFile).mapValues { (_, list) -> list.filter(methods::contains) }
```